biocrnpyler/components_basic.py

```diff
@@ -8,14 +8,41 @@
 from .reaction import Reaction
 from .species import Complex, Species
 
+class LengthComponent(Component):
+    """
+    A high level class for Components with Length
+    """
+    def __init__(self, name, length=0, **keywords):
+        """Initialize a DNA object to store DNA related information.
+
+        :param name: Name of the sequence (str)
+        :param length: length of the basepairs (int)
+        :param attributes: Species attribute
+        :param keywords: pass into the parent's (Component) initializer
+        """
+        self.length = length
+        Component.__init__(name = name, self=self, **keywords)
+
+    @property
+    def length(self):
+        return self._length
+
+    @length.setter
+    def length(self, length):
+        if length is None:
+            self._length = length
+        elif isinstance(length, int) and length >= 0:
+            self._length = length
+        else:
+            raise ValueError("Length must be a non-negative integer!")
 
-class DNA(Component):
+class DNA(LengthComponent):
```

Comment:
i guess `DNA_part` should be a `LengthComponent` too? I don't think it should be a `DNA` though

Reply:
Sure, I was unaware of LengthComponents but that makes sense